components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/action/PreIssueAccessTokenResponseProcessor.java

```diff
@@ -154,14 +163,56 @@ private void logOperationExecutionResults(ActionType actionType,
     }
 
     @Override
-    public ActionExecutionStatus processErrorResponse(Map<String, Object> map, Event event,
+    public ActionExecutionStatus<Failure> processFailureResponse(Map<String, Object> eventContext,
+                                                        Event actionEvent,
+                                                        ActionInvocationFailureResponse failureResponse) throws
+            ActionExecutionResponseProcessorException {
+
+        processServerError(failureResponse.getFailureReason());
```

Comment:
Failure response is to denote validations and to break explicitly. So it should not be processing server errors

Reply:
Given that oauth handles specific errors and send all other errors which are not server errors as '400' we discussed that this should simply populate the fauilre reason and the description coming right ?

Reply:
Is this to validate if the message is of format server error and to skip ?
The method name and diagnostics logs should change to give the right meaning then

Reply:
This is fixed.